src/python.c

```diff
@@ -510,6 +510,35 @@ static int cpy_notification_callback(const notification_t *notification,
   Notification *n;
 
   CPY_LOCK_THREADS
+  PyObject *dict = PyDict_New(); /* New reference. */
+  for (notification_meta_t *meta = notification->meta;
+       meta != NULL; meta = meta->next) {
+    PyObject *temp = NULL;
+    if (meta->type == NM_TYPE_STRING) {
+      temp = cpy_string_to_unicode_or_bytes(meta->nm_value.nm_string); /* New reference. */
+      PyDict_SetItemString(dict, meta->name, temp);
+      Py_XDECREF(temp);
+    } else if (meta->type == NM_TYPE_SIGNED_INT) {
+      PyObject *sival = PyLong_FromLongLong(meta->nm_value.nm_signed_int);
+      temp = PyObject_CallFunctionObjArgs((void *)&SignedType, sival,
```

Comment:
Why can't you just do this:?

```c
PyDict_SetItemString(dict, meta->name,
                     PyLong_FromLongLong(meta->nm_value.nm_signed_int));
```

Dito for `NM_TYPE_UNSIGNED_INT` below.

Reply:
It's possible but collectd python plugin doesn't use `long` type directly. It uses collectd python types (`Signed` and `Unsigned`) instead. Thus, we have to convert python long/unsigned long type into `Signed`/`Unsigned` types. Same has been done for python value.
https://github.com/collectd/collectd/blob/master/src/collectd-python.pod

Reply:
That makes sense, thank you! I didn't realize right away that the `Signed` / `Unsigned` types are *not* built-in literals but classes.